Framework/PythonInterface/plugins/algorithms/SNSPowderReduction.py

```diff
@@ -478,7 +478,28 @@ def PyExec(self):  # noqa
             # process the container
             can_run_numbers = self._info["container"].value
             can_run_numbers = ['%s_%d' % (self._instrument, value) for value in can_run_numbers]
-            can_run_ws_name = self._process_container_runs(can_run_numbers, samRunIndex, preserveEvents, absorptionWksp=a_container)
+            # Check if existing container
+            #  - has history
+            #    - was created using the same method
+            #       -> carry on as usual
+            #    - was created with a different method
+            #       -> delete the current one, then carry on
+            #  - no history (processing list of runs)
+            #    -> carry on as a single call wiht list of runs ensures that same method is used.
+            can_run_ws_name, _ = self._generate_container_run_name(can_run_numbers, samRunIndex)
+            if can_run_ws_name in mtd:
+                hstry = mtd[can_run_ws_name].getHistory()
+                if not hstry.empty():
+                    last_absMethod = hstry.getAlgorithm(0).getPropertyValue("TypeOfCorrection")
```

Comment:
I think you should check that the algorithm name is `SNSPowderReduction` as well. I don't think that any other algorithm has a `TypeOfCorrection` property, but adding the check would make it more thurough.

Reply:
the check on the algorithm name is added as suggested